src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -1857,23 +1871,23 @@ def forward(
 
         itc_loss = None
 
-        if labels is not None:
-            labels = torch.arange(len(labels), device=logits.device)
+        if return_loss:
+            labels = torch.arange(len(logits), device=logits.device)
             text_to_image_loss = nn.functional.cross_entropy(logits_text_to_image, labels)
             text_to_cross_loss = nn.functional.cross_entropy(logits_text_to_cross, labels)
             image_to_cross_loss = nn.functional.cross_entropy(logits_image_to_cross, labels)
             itc_loss = (text_to_image_loss + text_to_cross_loss + image_to_cross_loss) / 3.0
 
         if not return_dict:
-            output = tuple(logits)
+            output = (logits, text_embeds, image_embeds, cross_embeds) + outputs[3:]
             return ((itc_loss,) + output) if itc_loss is not None else output
 
         return BridgeTowerContrastiveOutput(
-            attentions=outputs.attentions,
-            hidden_states=outputs.hidden_states,
+            loss=itc_loss,
+            logits=logits,
             text_embeds=text_embeds,
             image_embeds=image_embeds,
             cross_embeds=cross_embeds,
-            logits=logits,
-            loss=itc_loss,
+            attentions=outputs.attentions,
+            hidden_states=outputs.hidden_states,
```

Comment:
```suggestion
            hidden_states=outputs.hidden_states,
            attentions=outputs.attentions,            
```

Reply:
The tests (currently disabled) all pass (launched manually). Good to merge once the above change is committed.

Reply:
Let me commit directly @tileintel as it's minor. Hope you are OK with it.